test/e2e/framework.go

```diff
@@ -954,13 +954,13 @@ func (data *TestData) createPodOnNode(name string, ns string, nodeName string, i
 	// image could be a fully qualified URI which can't be used as container name and label value,
 	// extract the image name from it.
 	imageName := getImageName(image)
-	return data.createPodOnNodeInNamespace(name, ns, nodeName, imageName, image, command, args, env, ports, hostNetwork, mutateFunc)
+	return data.createPodOnNodeInNamespace(name, ns, nodeName, imageName, image, command, args, env, ports, hostNetwork, mutateFunc, imageName)
```

Comment:
I don't think this should be injected like this. I think the caller should provide the desired labels through the `mutateFunc` parameter.

Reply:
Yes. Changed it back.